core/src/test/scala/org/apache/spark/sql/delta/stats/DataSkippingDeltaTests.scala

```diff
@@ -565,16 +576,61 @@ trait DataSkippingDeltaTestsBase extends QueryTest
       "a IS NULL",
       "a IS NOT NULL",
       "a = NULL", // Ideally this should not hit as it is always FALSE, but its correct to not skip
-      "a <=> NULL",
+      "NOT a = NULL", // Same as previous case
+      "a <=> NULL", // This is optimized to `IsNull(a)` by NullPropagation
+      "NOT a <=> NULL", // This is optimized to `IsNotNull(a)`
       "a = 1",
+      "a <=> 1",
       "TRUE",
       "FALSE",    // Ideally this should not hit, but its correct to not skip
       "NULL AND a = 1" // This is optimized to FALSE by ReplaceNullWithFalse, so it's same as above
     ),
     misses = Seq(
       "a <> 1",
       "a > 1",
-      "a < 1"
+      "a < 1",
+      "NOT a = 1",
+      "NOT a <=> 1"
+    )
+  )
+
+  testSkipping(
+    "nulls - EqualNullSafe(_, Literal(null, _)) always hit when only not-null in the file",
+    """
+      |{"a": "1"}
+      |""".stripMargin,
+    schema = new StructType().add(new StructField("a", StringType)),
+    hits = Seq(
+      "a <=> NULL",
+      "NOT a <=> NULL",
+      "a = NULL",
+      "a != NULL"
+    ),
+    misses = Seq.empty,
+    sqlConfs = Seq(
+      "spark.sql.optimizer.excludedRules" ->
+        ("org.apache.spark.sql.catalyst.optimizer.NullPropagation," +
```

Comment:
Thanks for writing these two `EqualNullSafe` tests. It makes me understand why we will never `null` cases.

We can remove these tests. It looks weird to have them here.